nmagent/cachedclient.go

```diff
@@ -0,0 +1,106 @@
+package nmagent
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+const GetHomeAzAPIName = "GetHomeAz"
+
+type CachedClient struct {
+	*Client
+	// The purpose of nmaSupportedApisCache here is only for reducing the call to nmagent when the getHomeAz api is supported,
+	// but having other unexpected errors while querying homeAz During the retry
+	nmaSupportedApisCache    []string
+	homeAzResponseCache      HomeAzResponse
+	homeAzResponseErrorCache error
```

Comment:
I don't think it's a good idea to cache errors, because it's more likely that we'll be telling lies to other parts of the system about the state of NMAgent. The same could be said about successes, of course, but at least we know the degree to which those lies extend.

Continuing to say that some dependency is broken, even after it's been fixed just perpetuates the brokenness and allows it to haunt those on-call.